e2e/ctl_v3_member_test.go

```diff
@@ -110,3 +113,110 @@ func ctlV3MemberUpdate(cx ctlCtx, memberID, peerURL string) error {
 	cmdArgs := append(cx.PrefixArgs(), "member", "update", memberID, fmt.Sprintf("--peer-urls=%s", peerURL))
 	return spawnWithExpect(cmdArgs, " updated in cluster ")
 }
+
+// TestCtlV3MemberAddRemove tests cluster member add,remove
+// and makes sure removed member cannot join the cluster and exits "in time".
+// (Related https://github.com/coreos/etcd/issues/7512)
+func TestCtlV3MemberAddRemove(t *testing.T) {
```

Comment:
this isn't really a ctl test, probably belongs in a separate file like `etcd_reconfig_test.go`; alternatively, it might make more sense as an integration test instead of e2e (i.e., just check that the removed member posts to StopNotify())